### PR TITLE
DENG-1021 Added destination table param to query commands

### PR DESCRIPTION
**Pull request by wwyc**
[DENG-1021](https://mozilla-hub.atlassian.net/browse/DENG-1021): 

* As part of implementing the managed backfill process, backfilled data are to be first stored in a separate `destination table`.
* This PR updates the query commands to allow `deploy` and `backill` cli commands to specify a destination table.

[DENG-1021]: https://mozilla-hub.atlassian.net/browse/DENG-1021?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bigquery_etl/cli/query.py [hunk 1]
```diff
@@ -862,6 +877,13 @@ def _run_query(
         if not use_public_table and destination_table is not None:
             # destination table was parsed by argparse, however if it wasn't modified to
             # point to a public table it needs to be passed as parameter for the query
+
+            if re.match(QUALIFIED_TABLE_NAME_RE, destination_table):
+                project, dataset, table = qualified_table_name_matching(
+                    destination_table
+                )
+                destination_table = "{}:{}.{}".format(project, dataset, table)
```
Comment (scholtzan) [inline bigquery_etl/cli/query.py:885]:
```suggestion
                destination_table = "{}.{}.{}".format(project, dataset, table)
```

usually we use the `project.dataset.table` notation for fully qualified tables

Comment (wwyc) [reply bigquery_etl/cli/query.py:885]: To pass `destination_table` into `query_arguments`, I think it needs the `{}:{}.{}` notation.  See https://github.com/mozilla/bigquery-etl/blob/28b06a9dd45fc76b5e12687b264da21de0f571df/bigquery_etl/cli/query.py#L857

File: bigquery_etl/cli/query.py [hunk 2]
```diff
@@ -1686,6 +1708,15 @@ def _update_query_schema(
     default=False,
     is_flag=True,
 )
+@click.option(
+    "--destination_table",
+    "--destination-table",
+    required=False,
+    help=(
+        "Destination table name results are written to. "
+        + "If not set, determines destination table based on query."
```
Comment (ANich) [inline bigquery_etl/cli/query.py:1717]: +"Must be fully qualified (project.dataset.table)" ?

File: bigquery_etl/cli/query.py (hunk 1 above)
Comment (ANich) [reply bigquery_etl/cli/query.py:885]: A voice in the distance whispers: "we should migrate to the bigquery API..." 😄

**Review by ANich — APPROVED** [review]
Overall LGTM, left two comments. Would be nice in the future if we could to having all destinations be fully qualified for clarity.

File: bigquery_etl/cli/query.py [hunk 3]
```diff
@@ -1778,7 +1811,10 @@ def deploy(
             dataset_name = query_file_path.parent.parent.name
             project_name = query_file_path.parent.parent.parent.name
 
-            full_table_id = f"{project_name}.{dataset_name}.{table_name}"
+            if destination_table:
+                full_table_id = destination_table
```
Comment (kik-kik) [inline bigquery_etl/cli/query.py:1815]: Looks like there's an assumption made here about what the user should enter in this param, `{project_name}.{dataset_name}.{table_name}` if this indeed is the case then perhaps we should validate the input matches this expectation.

**Review by kik-kik — APPROVED** [review]
Looks good to me, just make sure to address Anna's comments before merging. 🙇

**Review by scholtzan — APPROVED** [review]

File: bigquery_etl/cli/query.py (hunk 3 above)
Comment (wwyc) [reply bigquery_etl/cli/query.py:1846]: I'd take note to address this in a future iteration.

**Pull request merged**